lib/pacemaker/libpacemaker_private.h

```diff
@@ -16,6 +16,80 @@
 
 #include <crm/pengine/pe_types.h> // pe_action_t, pe_node_t, pe_working_set_t
 
+// Resource allocation methods
+struct resource_alloc_functions_s {
+    GHashTable *(*merge_weights) (pe_resource_t *, const char *, GHashTable *, const char *, float,
+                                  enum pe_weights);
+    pe_node_t *(*allocate) (pe_resource_t *, pe_node_t *, pe_working_set_t *);
+    void (*create_actions) (pe_resource_t *, pe_working_set_t *);
+    gboolean(*create_probe) (pe_resource_t *, pe_node_t *, pe_action_t *, gboolean, pe_working_set_t *);
```

Comment:
We might as well shorten the line lengths and make parenthesis spacing consistent for pointer-to-func return types, since all these lines will already be in the diff.

We could also add names to prototypes now if we want, though it looks like the usual approach is to wait until a doc comment update.

Reply:
Yep, all this will be in later PRs as I go through them one by one